cli/compose/interpolation/interpolation.go

```diff
@@ -38,9 +39,13 @@ func interpolateSectionItem(
 	for key, value := range item {
 		interpolatedValue, err := recursiveInterpolate(value, mapping)
 		if err != nil {
+			hint := ""
+			if strings.Count(err.Template, "$") != 2*strings.Count(err.Template, "$$") {
```

Comment:
I'm not sure about this. This will be `true` when there are valid variables as well.

Why don't we just include the hint in all cases? I don't know of any way to detect an incorrect usage.